File: Template Method/Golang/TwoFactorAuthentication.go
package main

type ITwoFactorAuthentication interface {
	getData(string)
	sendNotification(string)
	match(string)
}

type TwoFactorAuthentication struct {
	twoFactor ITwoFactorAuthentication
}

func (twoFactorAuthentication *TwoFactorAuthentication) authenticate(username, password string)  {
	twoFactorAuthentication.twoFactor.getData(username)

	twoFactorAuthentication.twoFactor.sendNotification(username)

	twoFactorAuthentication.twoFactor.match(username)
}

File: Template Method/PHP/TwoFactorAuthentication.php
<?php

abstract class TwoFactorAuthentication {
    abstract function getData(string $username);
    abstract function sendNotification(string $username);
    abstract function match(string $username);

    public function authenticate(string $username, string $password) {
        $this->getData($username);

        $this->sendNotification($username);

        $this->match($username);
    }
}